cobaya/likelihoods/planck_2018_lowl/TT_native.py

```diff
@@ -0,0 +1,157 @@
+            self.data['cl2x'][:, 2:, 2] = np.loadtxt(os.path.join(path, 'cl2x_3.txt')) 
+            self.data['mu'][2:] = np.loadtxt(os.path.join(path, 'mu.txt'))
+            self.data['mu_sigma'][2:] = np.loadtxt(os.path.join(path, 'mu_sigma.txt'))
+            _nbins = len(self.data['cl2x'][:, 0, 0])
+
+            # Bandlimit covariance matrix
+            for l in range(self._lmin, self._lmax+1):
+                for k in range(self._lmin, self._lmax+1):
+                    if abs(l-k) > self._delta_l:
+                        self.data[cov][l, k] = 0
+
+            # Set up prior
+            self.data['prior'] = np.zeros((251, 2))
+            for l in range(self._lmin, self._lmax+1):
+                j = 0
+                while abs(self.data['cl2x'][j, l, 1] + 5) < 1e-4:
+                    j += 1
+                self.data['prior'][l, 0] = self.data['cl2x'][j+2, l, 0]
+                j = _nbins-1
+                while abs(self.data['cl2x'][j, l, 1] - 5) < 1e-4:
+                    j -= 1
+                self.data['prior'][l, 1] = self.data['cl2x'][j-2, l, 0]
+
+            np.savetxt('prior.txt', self.data['prior'])
+
+            self.data['offset'] = self.log_likelihood(self.data['mu_sigma'], init=True)
+                
+
+    def get_requirements(self):
+        return {'Cl': {'tt': self._lmax}}
+
+
+    def log_likelihood(self, cls_TT, init=False):
+
+        if (any(cls_TT[self._lmin:self._lmax+1] < self.data['prior'][self._lmin:self._lmax+1, 0]) or
+                any(cls_TT[self._lmin:self._lmax+1] > self.data['prior'][self._lmin:self._lmax+1, 1])):
+            return -1e30
+
+        # Convert the cl's to Gaussianized variables
+        x = np.zeros(self._lmax+1)
+        for l in range(self._lmin, self._lmax+1):
+            x[l] = self._splint_gauss_br(self.data['cl2x'][:, l, 0],
+                                        self.data['cl2x'][:, l, 1],
+                                        self.data['cl2x'][:, l, 2],
+                                        cls_TT[l])
+        logl = -0.5 * sum((x[self._lmin:self._lmax+1] - self.data['mu'][self._lmin:self._lmax+1]) * 
+                          np.dot(self.data['cov'][self._lmin:self._lmax+1, self._lmin:self._lmax+1],
+                                 (x[self._lmin:self._lmax+1] - self.data['mu'][self._lmin:self._lmax+1])))
+
+        # Add Jacobian term
+        for l in range(self._lmin, self._lmax+1):
+            dxdCl = self._splint_deriv_gauss_br(self.data['cl2x'][:, l, 0],
+                                               self.data['cl2x'][:, l, 1],
+                                               self.data['cl2x'][:, l, 2],
+                                               cls_TT[l])
+            if dxdCl < 0:
+                return -1e30
+            else:
+                logl += np.log(dxdCl)
+        if not init:
+            logl -= self.data['offset']
+
+        return logl
+
+    def _splint_gauss_br(self, xa, ya, y2a, x):
```

Comment:
Use standard scipy interpolate functions instead? (presumably faster and neater)

Reply:
Yes, I would have liked to do this, it just was a bit more work at the moment because I have to figure out whether the original data files 'match' the scipy interpolate functions, as I cannot recreate the data files. But I suppose I could recreate them using the original splinting functions.